test/mitmproxy/proxy/test_mode_servers.py

```diff
@@ -98,6 +98,42 @@ async def test_tcp_start_stop(caplog_async):
         assert await caplog_async.await_log("stopped")
 
 
+async def test_tcp_timeout(caplog_async):
+    """Test that TCP connections are closed after the configured timeout period."""
+    caplog_async.set_level("INFO")
+    manager = MagicMock()
+
+    with taddons.context() as tctx:
+        # Set a very short timeout for testing (1 second)
+        tctx.options.tcp_timeout = 1
+
+        inst = ServerInstance.make("regular@127.0.0.1:0", manager)
+        await inst.start()
+        assert await caplog_async.await_log("proxy listening")
+
+        host, port, *_ = inst.listen_addrs[0]
+        reader, writer = await asyncio.open_connection(host, port)
+        assert await caplog_async.await_log("client connect")
+
+        # Keep connection open but inactive - it should timeout after 1s
+        # Wait a bit longer than the timeout to ensure it triggers
+        await asyncio.sleep(1.2)
```

Comment:
Remove - the `await_log` statement below waits for the log message to appear anyhow.
```suggestion
```